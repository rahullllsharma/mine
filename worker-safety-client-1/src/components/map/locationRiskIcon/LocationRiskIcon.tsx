import type { RiskLevel } from "@/components/riskBadge/RiskLevel";
import cx from "classnames";
import { Icon } from "@urbint/silica";
import {
  getDefaultTextColorByRiskLevel,
  getLocationRiskIconBackgroundColorByRiskLevel,
  getRiskIcon,
} from "@/utils/risk";

export type LocationRiskIconProps = {
  riskLevel: RiskLevel;
  label: string;
  isActive?: boolean;
  isCritical?: boolean;
};

export default function LocationRiskIcon({
  riskLevel,
  label,
  isActive = false,
  isCritical = false,
}: LocationRiskIconProps): JSX.Element {
  const iconName = getRiskIcon(riskLevel) ?? "help_questionmark";
  return (
    <div
      className={cx(
        "flex justify-center items-center w-auto min-w-[1.25rem] h-5 text-base rounded-full shadow-10 hover:ring hover:ring-white",
        getLocationRiskIconBackgroundColorByRiskLevel(riskLevel),
        getDefaultTextColorByRiskLevel(riskLevel),
        {
          "ring ring-white": isActive,
        }
      )}
      role="status"
      aria-label={label}
    >
      <Icon name={iconName} />
      {isCritical && (
        <Icon
          name={"warning"}
          className={cx(
            "pointer-events-none text-lg bg-transparent mx-1 leading-none",
            getDefaultTextColorByRiskLevel(riskLevel)
          )}
        />
      )}
    </div>
  );
}
