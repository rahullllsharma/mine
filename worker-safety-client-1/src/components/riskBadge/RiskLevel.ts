export enum RiskLevel {
  HIGH = "HIGH",
  MEDIUM = "MEDIUM",
  LOW = "LOW",
  UNKNOWN = "UNKNOWN",
  RECALCULATING = "RECALCULATING",
}
