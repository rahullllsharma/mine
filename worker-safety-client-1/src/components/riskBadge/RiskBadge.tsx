import { Badge, Icon } from "@urbint/silica";
import cx from "classnames";
import {
  getBackgroundColorByRiskLevel,
  getDefaultTextColorByRiskLevel,
  getRiskIcon,
} from "@/utils/risk/";
import { RiskLevel } from "./RiskLevel";

interface RiskBadgeAnimatedProps {
  label: string;
  isCritical?: boolean;
}

const RiskBadgeAnimated = ({
  label,
  isCritical = false,
}: RiskBadgeAnimatedProps): JSX.Element => {
  return (
    <div
      className="flex items-center font-bold shadow-5 h-5 rounded-full px-1.5 py-0.5 w-max uppercase whitespace-nowrap bg-risk-recalculating text-neutral-shade-75"
      role="note"
    >
      {/*<div
        className="animate-spin inline-block border-2 border-r-neutral-shade-75 w-3 h-3 mr-1 rounded-full border-neutral-shade-18"
        role="status"
      />*/}
      <span className="text-tiny m-1">{label}</span>
      {isCritical && (
        <Icon
          name={"warning"}
          className={cx(
            "pointer-events-none text-lg bg-transparent ml-1 leading-none"
          )}
        />
      )}
    </div>
  );
};

export interface RiskBadgeProps {
  risk: RiskLevel;
  label: string;
  isCritical?: boolean;
}

export default function RiskBadge({
  risk,
  label,
  isCritical = false,
}: RiskBadgeProps): JSX.Element {
  if (risk === RiskLevel.RECALCULATING) {
    return (
      <RiskBadgeAnimated label={label?.toUpperCase()} isCritical={isCritical} />
    );
  }

  return (
    <Badge
      className={cx(
        "whitespace-nowrap",
        getBackgroundColorByRiskLevel(risk),
        getDefaultTextColorByRiskLevel(risk)
      )}
      iconStart={getRiskIcon(risk)}
      label={label?.toUpperCase()}
      iconEnd={isCritical ? "warning" : undefined}
    />
  );
}
