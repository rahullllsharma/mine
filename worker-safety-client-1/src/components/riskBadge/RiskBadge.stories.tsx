import type { ComponentMeta, ComponentStory } from "@storybook/react";

import RiskBadge from "./RiskBadge";
import { RiskLevel } from "./RiskLevel";

export default {
  title: "Components/Badge/RiskBadge",
  component: RiskBadge,
  argTypes: {
    riskColor: { table: { disable: true } },
  },
} as ComponentMeta<typeof RiskBadge>;

const TemplateRiskBadge: ComponentStory<typeof RiskBadge> = ({
  risk,
  label,
}) => <RiskBadge risk={risk} label={label} />;

export const Playground = TemplateRiskBadge.bind({});

Playground.args = {
  risk: RiskLevel.HIGH,
  label: "Risk Label",
};
