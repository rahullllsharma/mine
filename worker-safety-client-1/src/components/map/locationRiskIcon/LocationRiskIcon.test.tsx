import { render, screen } from "@testing-library/react";
import { RiskLevel } from "@/components/riskBadge/RiskLevel";
import LocationRiskIcon from "./LocationRiskIcon";

describe(LocationRiskIcon.name, () => {
  it("should render correctly", () => {
    const { asFragment } = render(
      <LocationRiskIcon riskLevel={RiskLevel.HIGH} label="title" />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("should display the title correctly", () => {
    render(<LocationRiskIcon riskLevel={RiskLevel.HIGH} label="title" />);
    screen.getByRole("status", {
      name: /title/i,
    });
  });
});
