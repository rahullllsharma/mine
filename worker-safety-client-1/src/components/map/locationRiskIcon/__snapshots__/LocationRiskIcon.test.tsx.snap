// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationRiskIcon should render correctly 1`] = `
<DocumentFragment>
  <div
    aria-label="title"
    class="flex justify-center items-center w-auto min-w-[1.25rem] h-5 text-base rounded-full shadow-10 hover:ring hover:ring-white bg-risk-high text-white"
    role="status"
  >
    <i
      aria-hidden="true"
      class="ci-chevron_up"
    />
  </div>
</DocumentFragment>
`;
